controllers/remote/cluster.go

```diff
@@ -18,53 +18,41 @@ package remote
 
 import (
 	"github.com/pkg/errors"
-	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 	restclient "k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
 	kcfg "sigs.k8s.io/cluster-api/util/kubeconfig"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-// ClusterClient is an interface encapsulating methods
-// to access a remote cluster.
-type ClusterClient interface {
-	RESTConfig() *restclient.Config
-	CoreV1() (corev1.CoreV1Interface, error)
-}
-
-// clusterClient is a helper struct to connect to remote workload clusters.
-type clusterClient struct {
-	restConfig *restclient.Config
-	cluster    *clusterv1.Cluster
-}
+// NewClusterClient returns a Client for interacting with a remote Cluster
+func NewClusterClient(c client.Client, cluster *clusterv1.Cluster, scheme *runtime.Scheme) (client.Client, error) {
```

Comment:
Why pass in a custom scheme?

Reply:
Because this is for arbitrary consumers, and if their workload clusters have `Widgets` or `Frombles` or whatever, they need to be able to pass in a scheme that knows about them, to be able to encode/decode them.